### DISCUSSION TITLE
Can't get Timeouts to work

### DISCUSSION THREAD

**Discussion by thomasplarsson** [encode/databases#577] [Potential issue]
Hello.
I have a problem with timeouts not being raised as expected. If I pause a database (as in, pausing a postgres docker) and run an execute statement with a 5 sec timeout, this doesn't raise an exception until I unpause the database. The corresponding code with pure asyncpg works as expected. I'm wondering if I'm missing something obvious here.

My reproduce.py script:

```
import asyncio

import asyncpg
from databases import Database

async def main():
    url = f"postgresql://postgres:postgrespass@localhost:5432/postgres"
    db = Database(url=url, timeout=5, command_timeout=5)
    await db.connect()
    try:
        row = await db.execute(query="SELECT 'test'") # <-- pause the docker database before running this line
    except Exception as exc:
        print(f"Exception encountered: {exc=}, {type(exc)=}") # <-- Never gets here until database is unpaused.


    conn = await asyncpg.connect(url, timeout=5, command_timeout=5)
    try:
        row = await conn.fetchrow("SELECT 'test'") # <-- pause the docker database before running this line
    except Exception as exc:
        print(f"Exception encountered: {exc=}, {type(exc)=}") # <-- Gets here after timeout, while db is still in pause.
    await conn.close()

if __name__ == "__main__":
    asyncio.run(main())
```

As far as I can see, those timeout properties are propagated down from the Database object to the underlying connection just as if I instantiate the connection myself. Feels like it something else. Has anyone else seen this and perhaps have a clue?

Thanks
/Thomas

**Comment by Charlesnorris509** [comment]
Hey @thomasplarsson 

Hi Thomas,

I know its been a while since you posted this issue, but it likely stems from how databases handles timeouts with its connection pool. Try disabling pooling by using a direct connection:


async with db.connection() as conn:
    try:
        row = await conn.execute(query="SELECT 'test'")
    except Exception as exc:
        print(f"Exception: {exc}")
